### DISCUSSION TITLE
Does Nuxt provide a way to persist state on the Server?

### DISCUSSION THREAD

**Discussion by MarioC3** [nuxt/nuxt#17362] [Questions]
Like the title says: Does Nuxt provide a way to persist state on the server?

I was checking out [Nitro Storage Layer with unstorage](https://nitro.unjs.io/guide/storage.html) to use it to persist. And even so, will it be only available on the server? Is that the only way?

**Comment by Flanker72** [comment]
What kind of state are you going to store?
Passing state between server and client may be implemented using `useState` approach during SSR.
Persistent state may be implemented using some kind of session store using middlewares and ssrContext.

**Reply by MarioC3** [reply]
API configs, API tokens, and other data we don't want to expose to the client. Basically, a data object.

We get the data (via an API call) when the app is created; I'm looking for a way to persist that data on the server-side and not pass it to the client.

**Reply by Flanker72** [reply]
It seems you just need to create some kind of init script, then fulfill the singleton/state and use it everywhere you need.
For init script usage please, refer to [my article, step 2](https://medium.com/@flanker72/nuxt3-complex-solutions-database-integration-8df941f0fb82).
